test/dict.jl

```diff
@@ -294,6 +294,18 @@ for d in (Dict("\n" => "\n", "1" => "\n", "\n" => "2"),
     @test !isempty(summary(values(d)))
 end
 
+# Issue #15739 - Compact REPL printouts of an `Associative` use brackets when appropriate
+let d = Dict((1=>2) => (3=>45), (3=>10) => (10=>11))
+    buf = IOBuffer()
+    showcompact(buf, d)
+
+    # Ordering differs on CPU bitness, so build up the output for comparision
+    # each time.
+    expectedstr = join([string(x) for x in d], ",")
+
+    @test bytestring(buf) == "Dict($expectedstr)"
```

Comment:
Why not check whether it contains both expected strings. If somebody were to say break printing of Pairs, this would still pass.